src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/mailActionFields.tsx

```diff
@@ -0,0 +1,113 @@
+import {IssueAlertRuleAction, MailActionTargetType} from 'app/types/alerts';
+import {PanelItem} from 'app/components/panels';
+import space from 'app/styles/space';
+
+type Props = {
+  project: Project;
+  organization: Organization;
+  disabled: boolean;
+  loading: boolean;
+  action: IssueAlertRuleAction;
+  onChange: (action: IssueAlertRuleAction) => void;
+};
+
+interface OptionRecord {
+  value: string;
+  label: string;
+}
+
+class MailActionFields extends React.Component<Props> {
+  handleChange = (attribute: 'targetType' | 'targetIdentifier', newValue: string) => {
+    const {onChange, action} = this.props;
+    if (newValue === action[attribute]) {
+      return;
+    }
+    const newAction = {
+      ...action,
+      [attribute]: newValue,
+    };
+    /**
+     * TargetIdentifiers between the targetTypes are not unique, and may wrongly map to something that has not been
+     * selected. E.g. A member and project can both have the `targetIdentifier`, `'2'`. Hence we clear the identifier.
+     **/
+    if (attribute === 'targetType') {
+      newAction.targetIdentifier = '';
+    }
+    onChange(newAction);
+  };
+
+  handleChangeActorType = (optionRecord: OptionRecord) => {
+    this.handleChange('targetType', optionRecord.value);
+  };
+
+  handleChangeActorId = (optionRecord: OptionRecord & {[key: string]: any}) => {
+    this.handleChange('targetIdentifier', optionRecord.value);
+  };
+
+  render(): React.ReactElement {
+    const {disabled, loading, project, organization, action} = this.props;
+
+    const isIssueOwners = action.targetType === MailActionTargetType.IssueOwners;
+    const isTeam = action.targetType === MailActionTargetType.Team;
+
+    return (
+      <PanelItemGrid>
+        <SelectControl
+          isClearable={false}
+          isDisabled={disabled || loading}
+          value={action.targetType}
+          styles={{
+            control: provided => ({
+              ...provided,
+              minHeight: '28px',
+              height: '28px',
+            }),
+          }}
+          options={[
+            {value: MailActionTargetType.IssueOwners, label: 'Issue Owners'},
+            {value: MailActionTargetType.Team, label: 'Team'},
+            {value: MailActionTargetType.Member, label: 'Member'},
+          ]}
+          onChange={this.handleChangeActorType}
+        />
+        {!isIssueOwners ? (
+          <SelectMembers
+            disabled={disabled}
+            key={isTeam ? MailActionTargetType.Team : MailActionTargetType.Member}
+            showTeam={isTeam}
+            project={project}
+            organization={organization}
+            // The value from the endpoint is of type `number`, `SelectMembers` require value to be of type `string`
+            value={`${action.targetIdentifier}`}
+            styles={{
+              control: provided => ({
+                ...provided,
+                minHeight: '28px',
+                height: '28px',
+              }),
+            }}
```

Comment:
This is duplicated once, would it be worth refactoring?

Reply:
Done, refactored (: